src/main/java/org/opentripplanner/routing/edgetype/BikeWalkableEdge.java

```diff
@@ -0,0 +1,68 @@
+package org.opentripplanner.routing.edgetype;
+
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.StateEditor;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.graph.Edge;
+
+public interface BikeWalkableEdge {
+
+    default boolean canSwitchToWalkingBike(State state) {
+        return state.getNonTransitMode() == TraverseMode.BICYCLE;
+    }
+
+    default void switchToWalkingBike(
+            RoutingRequest options,
+            StateEditor editor,
+            boolean includeCostAndTime
+    ) {
+        editor.setBackWalkingBike(true);
+        if (includeCostAndTime) {
+            editor.incrementWeight(options.bikeSwitchCost);
+            editor.incrementTimeInSeconds(options.bikeSwitchTime);
+        }
+    }
+
+    default void switchToBiking(RoutingRequest options, StateEditor editor) {
+        editor.setBackWalkingBike(false);
+        editor.incrementWeight(options.bikeSwitchCost);
+        editor.incrementTimeInSeconds(options.bikeSwitchTime);
+    }
+
+    default StateEditor createEditorForWalking(State state, Edge edge) {
```

Comment:
Seems like this method is used for ElevatorEdge and PathWayEdge. What is the reason the same methods cannot be used for those edges as for the StreetEdge?

Reply:
I've added a commit to combine the logic so that it is only handled in `BikeWalkableEdge`.